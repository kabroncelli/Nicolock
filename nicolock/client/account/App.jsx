import React from 'react'

import SignIn from './components/SignIn'
import Register from './components/Register'
import Forgot from './components/Forgot'

const AccountApp = (props) => {
  return (
    <div className='modal-container position-login'>
      <Register />
      <SignIn />
      <Forgot />
    </div>
  )
}

export default AccountApp
