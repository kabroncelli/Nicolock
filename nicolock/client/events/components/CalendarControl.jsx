import React from 'react'
import PropTypes from 'prop-types'

const CalendarControl = ({ backwardClick, forwardClick, month }) => {
  return (
    <div>
      <div className='calendar-control'>
        <div className='controls'>
          <a className='calendar-back' onClick={backwardClick} href='#' />
          <span>{month}</span>
          <a className='calendar-forward' onClick={forwardClick} href='#' />
        </div>
      </div>
    </div>
  )
}

CalendarControl.propTypes = {
  month: PropTypes.string.isRequired,
  backwardClick: PropTypes.func.isRequired,
  forwardClick: PropTypes.func.isRequired,
}

export default CalendarControl
