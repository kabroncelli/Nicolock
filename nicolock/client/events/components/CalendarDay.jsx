import React from 'react'
import PropTypes from 'prop-types'

const CalendarDay = ({ dayNumber, events }) => {
  return (
    <div className='day'>
      <span className='day-number'>{dayNumber}</span>
      <div className='day-events'>
        {events.map(event =>
          <a className='link neutral-0' href={`/events/${event.slug}`} key={event.id}>{event.name}</a>
        )}
      </div>
    </div>
  )
}

CalendarDay.propTypes = {
  dayNumber: PropTypes.number,
  events: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    slug: PropTypes.string,
  })),
}

export default CalendarDay
