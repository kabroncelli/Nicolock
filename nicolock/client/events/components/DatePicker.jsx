import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'

class Datepicker extends React.Component {
  static propTypes = {
    dateChange: PropTypes.func.isRequired,
    date: PropTypes.object.isRequired,
  }

  constructor (props) {
    super(props)
    this.state = this.getStateFromDate(props.date)
  }

  componentWillReceiveProps (props) {
    const date = props.date || moment()
    this.setState(this.getStateFromDate(date))
  }

  getStateFromDate (date) {
    return {
      monthInput: date.format('MM'),
      dayInput: date.format('DD'),
      yearInput: date.format('YYYY'),
      month: date.format('MM'),
      day: date.format('DD'),
      year: date.format('YYYY'),
    }
  }

  monthChange = (e) => {
    const stringEntered = e.target.value.replace(' ', '')
    if (stringEntered.length > 2) {
      return null
    }
    this.setState({
      monthInput: stringEntered,
    })
    const valueEntered = parseInt(stringEntered)
    if (isNaN(valueEntered)) {
      return null
    }
    if (valueEntered < 1 || valueEntered > 12) {
      return null
    }
    // If the input was actually valid we can save it
    this.setState({month: valueEntered})
  }

  dayChange = (e) => {
    const stringEntered = e.target.value.replace(' ', '')
    if (stringEntered.length > 2) {
      return null
    }
    this.setState({
      dayInput: stringEntered,
    })
    const valueEntered = parseInt(stringEntered)
    if (isNaN(valueEntered)) {
      return null
    }
    if (valueEntered < 1 || valueEntered > 31) {
      return null
    }
    // If the input was actually valid we can save it
    this.setState({day: valueEntered})
  }

  yearChange = (e) => {
    const stringEntered = e.target.value.replace(' ', '')
    if (stringEntered.length > 4) {
      return null
    }
    this.setState({
      yearInput: stringEntered,
    })
    const valueEntered = parseInt(stringEntered)
    if (isNaN(valueEntered)) {
      return null
    }
    // If the input was actually valid we can save it
    this.setState({year: valueEntered})
  }

  findEventsSubmit = (e) => {
    e.preventDefault()
    const newDate = moment(`${this.state.month}-${this.state.day}-${this.state.year}`, 'M-D-YYYY')
    this.props.dateChange(newDate)
    this.setState(this.getStateFromDate(newDate))
  }

  render () {
    return (
      <form onSubmit={this.findEventsSubmit} className='row picker-wrap'>
        <span className='text'>
          Search<br />by date
        </span>
        <div className='picker'>
          <input
            onChange={this.monthChange}
            value={this.state.monthInput}
            type='text' placeholder='--' /> /
          <input
            onChange={this.dayChange}
            value={this.state.dayInput}
            type='text'
            placeholder='--' /> /
          <input
            onChange={this.yearChange}
            value={this.state.yearInput}
            type='text' placeholder='--' />
        </div>
        <button className='btn btn-sm hollow'>Find events</button>
      </form>
    )
  }
}

export default Datepicker
