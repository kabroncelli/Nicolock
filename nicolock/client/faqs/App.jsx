import React from 'react'
import PropTypes from 'prop-types'
import helpers from './utils/helpers'

const FaqSelected = (props) => {
  const { question, answer, relatedQuestions } = props.faqSelected
  return (
    <div className='faq-wrap'>
      <div className='faq-answers'>
        <div className='question'>{question}</div>
        <div className='answer'>{answer}</div>
      </div>
      <div className='related-questions'>
        <p className='paragraph'>Related Questions:</p>
        <ul className='misc-menu faq-link'>
          {relatedQuestions ? relatedQuestions.map((question, index) => (
            <li key={`relatedQuestion-${index}`} >
              <span className='link faq neutral-0' onClick={question.selectQuestion}>{question.question}</span>
            </li>
          )) : null}
        </ul>
      </div>
    </div>
  )
}

FaqSelected.propTypes = {
  faqSelected: PropTypes.shape({
    question: PropTypes.string.isRequired,
    answer: PropTypes.string.isRequired,
    relatedQuestions: PropTypes.arrayOf(
      PropTypes.shape({
        question: PropTypes.string.isRequired,
        selectQuestion: PropTypes.func.isRequired,
      })
    ),
  }).isRequired,
}

class App extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      faqList: null,
      faqSelected: null,
    }
  }

  componentWillMount () {
    helpers.getFaqs()
      .then((faqs) => {
        const newFaqList = faqs.group(item => item.category_name)
        this.setState(() => {
          return {
            originalFaqList: faqs,
            faqList: newFaqList,
          }
        }, () => {
          if (newFaqList.length > 0) {
            this.handleChange(newFaqList[0].data[0])
          }
        })
      })
  }

  handleChange = (data) => {
    this.setState((previousState) => {
      return {
        faqSelected: {
          question: data.question,
          answer: data.answer,
          relatedQuestions: previousState.originalFaqList.filter(faq => data.related.includes(faq.id)).map(faq => {
            return {
              question: faq.question,
              selectQuestion: this.handleChange.bind(this, faq),
            }
          }),
        },
      }
    })
  }

  render () {
    return (
      <div className='trifold-content'>
        <div className='trifold-middle faq col-faq'>
          <h1 className='title'>FAQs</h1>
          {!this.state.faqList
            ? <p>Loading</p>
            : this.state.faqList.map((category) => {
              return (
                <div key={category.key}>
                  <p className='paragraph'>{category.key}:</p>
                  <ul className='misc-menu faq-link'>
                    {category.data.map((question, index) => {
                      return (
                        <li key={`question-${index}`}>
                          <span className='link faq neutral-0' onClick={() => this.handleChange(question)}>{question.question}</span>
                        </li>
                      )
                    })}
                  </ul>
                </div>
              )
            })}
        </div>
        <div className='trifold-data col-faq'>
          {!this.state.faqSelected
            ? <p>Loading</p>
            : <FaqSelected faqSelected={this.state.faqSelected} />}
        </div>
      </div>
    )
  }
}

export default App
