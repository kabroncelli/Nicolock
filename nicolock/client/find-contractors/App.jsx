import React from 'react'
import PropTypes from 'prop-types'

import Panel from './components/Panel'
import TextField from './components/TextField'
import Dropdown from '../components/Dropdown'
import mapHelpers from '../nico-map/utils/helpers'

class App extends React.Component {
  static propTypes = {
    resourceUrl: PropTypes.string.isRequired,
    authenticated: PropTypes.bool,
  }

  constructor (props) {
    super(props)

    this.state = {
      contractors: [],
      zipCode: '',
      radius: 20,
    }
  }

  getRadiusOptions () {
    return [
      5, 10, 15, 20, 25, 30, 35, 40, 45, 50,
    ]
  }

  zipCodeChanged = (value) => {
    this.setState({zipCode: value})
  }

  radiusChanged = (value) => {
    this.setState({radius: value})
  }

  getContractors = (e) => {
    e.preventDefault()
    if (this.state.zipCode !== '') {
      mapHelpers.getNearbyZipCodes(this.state.zipCode, this.state.radius)
        .then((zipCodes) => {
          mapHelpers.getItems(this.props.resourceUrl, zipCodes)
            .then((contractors) => {
              this.setState({contractors: contractors})
            })
        })
    }
  }

  renderContractor = (contractor) => {
    return (
      <Panel
        key={contractor.id}
        id={contractor.user_id}
        companyName={contractor.name}
        phoneNumber={contractor.phone}
        email={contractor.email}
        city={contractor.city}
        state={contractor.state}
        authenticated={this.props.authenticated}
        profileUrl={contractor.profile_url}
        certifications={contractor.certifications} />
    )
  }

  render () {
    return (
      <div className='trifold-content'>
        <div className='trifold-middle'>
          <h1 className='title'>Locate A Contractor</h1>
          <p className='subtitle'>
          Enter your zip code to find contractors in your area
          </p>
          <form onSubmit={this.getContractors} className='locate-contractor-form'>
            <TextField label='Enter zip' valueEntered={this.zipCodeChanged} />
            <Dropdown
              label='Radius'
              options={this.getRadiusOptions()}
              valueEntered={this.radiusChanged} />
            <div className='action-group'>
              <button
                className='btn btn-sm neutral-0'
                type='submit'>Go</button>
            </div>
          </form>
          {this.state.contractors.map(this.renderContractor)}
        </div>
      </div>
    )
  }
}

export default App
