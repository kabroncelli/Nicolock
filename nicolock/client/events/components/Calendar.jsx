import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'

import helpers from '../utils/helpers'
import CalendarDay from './CalendarDay'
import CalendarControl from './CalendarControl'

class Calendar extends React.Component {
  static propTypes = {
    date: PropTypes.object,
    dateChange: PropTypes.func,
  }

  constructor (props) {
    super(props)
    const date = this.props.date || moment()
    this.state = this.getStateFromDate(date)
    this.state.events = []
  }

  componentDidMount () {
    this.loadEventsForDate(this.state.date)
  }

  componentWillReceiveProps (props) {
    const date = props.date || moment()
    this.setState(this.getStateFromDate(date))
    this.loadEventsForDate(date)
  }

  loadEventsForDate (date) {
    helpers.getEvents(date.format('M'), date.format('YYYY')).then(events => {
      this.setState({events: events})
    })
  }

  getStateFromDate (date) {
    return {
      date: date,
      currentMonth: date.format('MMMM'),
      currentYear: date.format('YYYY'),
      weeks: this.getWeeks(date),
    }
  }

  getWeeks (date) {
    const firstDay = date.clone().startOf('month')
    const lastDay = date.clone().endOf('month')
    // Need to start counting from a sunday
    // use the weekday() function to find out how many
    // days from the previous month we need and
    // how many days from the next month
    firstDay.subtract(firstDay.weekday(), 'days')
    lastDay.add(6 - lastDay.weekday(), 'days')

    let currentDate = firstDay.clone()
    let weeks = []
    let currentRow = 0
    while (currentDate.isBefore(lastDay)) {
      weeks.push([])
      for (var i = 0; i < 7; i++) {
        weeks[currentRow].push(currentDate.clone())
        currentDate.add(1, 'days')
      }
      currentRow++
    }
    return weeks
  }

  moveCalendarForward = (e) => {
    e.preventDefault()
    const newDate = this.state.date.clone().endOf('month').add(1, 'days')
    this.setState(this.getStateFromDate(newDate))
    this.props.dateChange(newDate)
    this.loadEventsForDate(newDate)
  }

  moveCalendarBackward = (e) => {
    e.preventDefault()
    const newDate = this.state.date.clone().startOf('month').subtract(1, 'days')
    this.setState(this.getStateFromDate(newDate))
    this.props.dateChange(newDate)
    this.loadEventsForDate(newDate)
  }

  renderCalendar () {
    return (
      <div>
        <div className='calendar-header'>
          {moment.weekdays().map((day, index) =>
            <span key={index}>{day}</span>
          )}
        </div>
        {this.state.weeks.map((week, index) =>
          <div className='calendar-row' key={index}>
            {week.map(day =>
              <CalendarDay
                dayNumber={parseInt(day.format('D'))}
                events={this.state.events.filter(event => moment(event.start_date, 'YYYY-MM-DD').format('MDY') === day.format('MDY'))}
                key={day.format('MDY')} />
            )}
          </div>
        )}
      </div>
    )
  }

  render () {
    return (
      <div className='calendar'>
        <CalendarControl
          forwardClick={this.moveCalendarForward}
          backwardClick={this.moveCalendarBackward}
          month={this.state.currentMonth} />
        {this.renderCalendar()}
      </div>
    )
  }
}

export default Calendar
