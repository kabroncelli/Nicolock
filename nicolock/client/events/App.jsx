import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'

import DatePicker from './components/DatePicker'
import Calendar from './components/Calendar'

class App extends React.Component {
  static propTypes = {
    date: PropTypes.object,
  }

  constructor (props) {
    super(props)
    this.state = {
      date: props.date || moment(),
    }
  }

  dateChanged = (date) => {
    this.setState({date: date})
  }

  render () {
    return (
      <div>
        <div className='row'>
          <h1 className='title'>Events &amp;<br />Training</h1>
          <DatePicker
            date={this.state.date}
            dateChange={this.dateChanged} />
        </div>
        <Calendar
          date={this.state.date}
          dateChange={this.dateChanged} />
      </div>
    )
  }
}

export default App
