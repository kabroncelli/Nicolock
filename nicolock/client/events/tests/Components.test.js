import React from 'react'
import renderer from 'react-test-renderer'
import moxios from 'moxios'
import moment from 'moment'

import App from '../App'
import Calendar from '../components/Calendar'
import CalendarControl from '../components/CalendarControl'
import CalendarDay from '../components/CalendarDay'
import DatePicker from '../components/DatePicker'

describe('Event components render correctly', () => {
  beforeEach(() => {
    moxios.install()
    moxios.stubRequest(
      '/rest/events',
      {
        status: 200,
        response: [
          {
            'id': 1,
            'name': 'September Event',
            'description': 'Some event in september',
            'start_date': '2017-09-26',
            'start_time': '20:35:17',
            'end_date': '2017-09-26',
            'end_time': '20:35:19',
            'timezone': 'US/Mountain',
            'address': '570 N 300 E',
            'city': 'Logan',
            'state': 'UT',
            'postal_code': '84321',
            'register_link': '',
          },
        ],
      }
    )
  })

  afterEach(() => {
    moxios.uninstall()
  })

  it('App renders correctly', () => {
    const tree = renderer.create(
      <App date={moment('2017-09-26', 'YYYY-MM-DD')} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Calendar renders correctly', () => {
    const dateChangeFunc = jest.fn()
    const tree = renderer.create(
      <Calendar
        date={moment('2017-09-26', 'YYYY-MM-DD')}
        dateChange={dateChangeFunc} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('CalendarControl renders correctly', () => {
    const backwardFunc = jest.fn()
    const forwardFunc = jest.fn()
    const tree = renderer.create(
      <CalendarControl
        backwardClick={backwardFunc}
        forwardClick={forwardFunc}
        month={'September'} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('CalendarDay renders correctly', () => {
    const tree = renderer.create(
      <CalendarDay
        day={1}
        events={[{id: 1, name: 'Test Event', slug: 'Slug'}]} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('DatePicker renders correctly', () => {
    const dateChangeFunc = jest.fn()
    const tree = renderer.create(
      <DatePicker
        dateChange={dateChangeFunc}
        date={moment('2017-09-26', 'YYYY-MM-DD')} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
