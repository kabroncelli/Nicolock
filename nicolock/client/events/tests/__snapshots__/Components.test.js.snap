// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event components render correctly App renders correctly 1`] = `
<div>
  <div
    className="row"
  >
    <h1
      className="title"
    >
      Events &
      <br />
      Training
    </h1>
    <form
      className="row picker-wrap"
      onSubmit={[Function]}
    >
      <span
        className="text"
      >
        Search
        <br />
        by date
      </span>
      <div
        className="picker"
      >
        <input
          onChange={[Function]}
          placeholder="--"
          type="text"
          value="09"
        />
         /
        <input
          onChange={[Function]}
          placeholder="--"
          type="text"
          value="26"
        />
         /
        <input
          onChange={[Function]}
          placeholder="--"
          type="text"
          value="2017"
        />
      </div>
      <button
        className="btn btn-sm hollow"
      >
        Find events
      </button>
    </form>
  </div>
  <div
    className="calendar"
  >
    <div>
      <div
        className="calendar-control"
      >
        <div
          className="controls"
        >
          <a
            className="calendar-back"
            href="#"
            onClick={[Function]}
          />
          <span>
            September
          </span>
          <a
            className="calendar-forward"
            href="#"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
    <div>
      <div
        className="calendar-header"
      >
        <span>
          Sunday
        </span>
        <span>
          Monday
        </span>
        <span>
          Tuesday
        </span>
        <span>
          Wednesday
        </span>
        <span>
          Thursday
        </span>
        <span>
          Friday
        </span>
        <span>
          Saturday
        </span>
      </div>
      <div
        className="calendar-row"
      >
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            27
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            28
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            29
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            30
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            31
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            1
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            2
          </span>
          <div
            className="day-events"
          />
        </div>
      </div>
      <div
        className="calendar-row"
      >
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            3
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            4
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            5
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            6
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            7
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            8
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            9
          </span>
          <div
            className="day-events"
          />
        </div>
      </div>
      <div
        className="calendar-row"
      >
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            10
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            11
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            12
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            13
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            14
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            15
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            16
          </span>
          <div
            className="day-events"
          />
        </div>
      </div>
      <div
        className="calendar-row"
      >
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            17
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            18
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            19
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            20
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            21
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            22
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            23
          </span>
          <div
            className="day-events"
          />
        </div>
      </div>
      <div
        className="calendar-row"
      >
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            24
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            25
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            26
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            27
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            28
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            29
          </span>
          <div
            className="day-events"
          />
        </div>
        <div
          className="day"
        >
          <span
            className="day-number"
          >
            30
          </span>
          <div
            className="day-events"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Event components render correctly Calendar renders correctly 1`] = `
<div
  className="calendar"
>
  <div>
    <div
      className="calendar-control"
    >
      <div
        className="controls"
      >
        <a
          className="calendar-back"
          href="#"
          onClick={[Function]}
        />
        <span>
          September
        </span>
        <a
          className="calendar-forward"
          href="#"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div>
    <div
      className="calendar-header"
    >
      <span>
        Sunday
      </span>
      <span>
        Monday
      </span>
      <span>
        Tuesday
      </span>
      <span>
        Wednesday
      </span>
      <span>
        Thursday
      </span>
      <span>
        Friday
      </span>
      <span>
        Saturday
      </span>
    </div>
    <div
      className="calendar-row"
    >
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          27
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          28
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          29
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          30
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          31
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          1
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          2
        </span>
        <div
          className="day-events"
        />
      </div>
    </div>
    <div
      className="calendar-row"
    >
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          3
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          4
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          5
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          6
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          7
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          8
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          9
        </span>
        <div
          className="day-events"
        />
      </div>
    </div>
    <div
      className="calendar-row"
    >
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          10
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          11
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          12
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          13
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          14
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          15
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          16
        </span>
        <div
          className="day-events"
        />
      </div>
    </div>
    <div
      className="calendar-row"
    >
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          17
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          18
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          19
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          20
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          21
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          22
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          23
        </span>
        <div
          className="day-events"
        />
      </div>
    </div>
    <div
      className="calendar-row"
    >
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          24
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          25
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          26
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          27
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          28
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          29
        </span>
        <div
          className="day-events"
        />
      </div>
      <div
        className="day"
      >
        <span
          className="day-number"
        >
          30
        </span>
        <div
          className="day-events"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Event components render correctly CalendarControl renders correctly 1`] = `
<div>
  <div
    className="calendar-control"
  >
    <div
      className="controls"
    >
      <a
        className="calendar-back"
        href="#"
        onClick={[Function]}
      />
      <span>
        September
      </span>
      <a
        className="calendar-forward"
        href="#"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`Event components render correctly CalendarDay renders correctly 1`] = `
<div
  className="day"
>
  <span
    className="day-number"
  />
  <div
    className="day-events"
  >
    <a
      className="link neutral-0"
      href="/events/Slug"
    >
      Test Event
    </a>
  </div>
</div>
`;

exports[`Event components render correctly DatePicker renders correctly 1`] = `
<form
  className="row picker-wrap"
  onSubmit={[Function]}
>
  <span
    className="text"
  >
    Search
    <br />
    by date
  </span>
  <div
    className="picker"
  >
    <input
      onChange={[Function]}
      placeholder="--"
      type="text"
      value="09"
    />
     /
    <input
      onChange={[Function]}
      placeholder="--"
      type="text"
      value="26"
    />
     /
    <input
      onChange={[Function]}
      placeholder="--"
      type="text"
      value="2017"
    />
  </div>
  <button
    className="btn btn-sm hollow"
  >
    Find events
  </button>
</form>
`;
